frontend/packages/console-app/src/components/volume-snapshot/create-volume-snapshot/create-volume-snapshot.tsx

```diff
@@ -0,0 +1,344 @@
+  }, [pvcSC]);
+
+  const filter = (snapshotClass: VolumeSnapshotClassKind) =>
+    !scObj || !scObj.provisioner ? false : scObj?.provisioner.includes(snapshotClass?.driver);
+
+  if (scError) {
+    return (
+      <Alert
+        className="co-alert co-volume-snapshot__alert-body"
+        variant="danger"
+        title="Error fetching info on claim's provisioner"
+        isInline
+      />
+    );
+  }
+
+  return (
+    <ListDropdown
+      {...props}
+      desc="Volume Snapshot Class with same provisioner as claim"
+      dataFilter={filter}
+      resources={resources}
+      selectedKeyKind={kind}
+      placeholder="Select snapshot class"
+      selectedKey={selectedKey}
+    />
+  );
+};
+
+const PVCSummary: React.FC<PVCSummaryProps> = ({ persistentVolumeClaim }) => {
+  const accessModeLabels = Object.freeze({
+    ReadWriteOnce: 'Single User (RWO)',
+    ReadWriteMany: 'Shared Access (RWX)',
+    ReadOnlyMany: 'Read Only (ROX)',
+  });
+
+  const storageClass = persistentVolumeClaim?.spec?.storageClassName;
+  const requestedCapacity = persistentVolumeClaim?.spec?.resources?.requests?.storage;
+  const sizeBase = convertToBaseValue(requestedCapacity);
+  const sizeMetrics = requestedCapacity ? humanizeBinaryBytes(sizeBase).string : '-';
+  const accessModes = accessModeLabels[persistentVolumeClaim?.spec?.accessModes];
+  const volumeMode = persistentVolumeClaim?.spec?.volumeMode;
+  return (
+    <dl>
+      <dt className="co-volume-snapshot__details-body">
+        {PersistentVolumeClaimModel.label} Details
+      </dt>
+      <dt>Name</dt>
+      <dd>
+        <ResourceIcon kind={PersistentVolumeClaimModel.kind} />
+        {getName(persistentVolumeClaim)}
+      </dd>
+      <dt>Namespace</dt>
+      <dd>
+        <ResourceIcon kind={NamespaceModel.kind} />
+        {getNamespace(persistentVolumeClaim)}
+      </dd>
+      <dt>Status</dt>
+      <dd>
+        <PVCStatus pvc={persistentVolumeClaim} />
+      </dd>
+      <dt>Storage Class</dt>
+      <dd>
+        <ResourceIcon kind={StorageClassModel.kind} />
+        {storageClass}
+      </dd>
+      <dt>Requested Capacity</dt>
+      <dd>{sizeMetrics}</dd>
+      <dt>Access Mode</dt>
+      <dd>{accessModes}</dd>
+      <dt>Volume Mode</dt>
+      <dd>{volumeMode}</dd>
+    </dl>
+  );
+};
+
+const CreateSnapshotForm = withHandlePromise<SnapshotResourceProps>((props) => {
+  const { resourceName, namespace, kindObj, handlePromise, inProgress, errorMessage } = props;
+
+  const [pvcName, setPVCName] = React.useState(resourceName);
+  const [pvcObj, setPVCObj] = React.useState(null);
+  const [snapshotName, setSnapshotName] = React.useState(`${pvcName || 'pvc'}-snapshot`);
+  const [snapshotClassName, setSnapshotClassName] = React.useState('');
+  const title = 'Create VolumeSnapshot';
+
+  const resourceWatch = React.useMemo(() => {
+    return Object.assign(
+      {
+        kind: PersistentVolumeClaimModel.kind,
+        namespace,
+        isList: true,
```

Comment:
Shouldn't isList be `true` or `false` based on pvcName is present or not as well?